controllers/auth.go

```diff
@@ -97,7 +97,53 @@ func (c *ApiController) Login() {
 		panic(err)
 	}
 
-	if form.Username != "" {
+	if form.Email != "" && form.EmailCode != "" {
+		var verificationCodeType string
+
+		// check result through Email or Phone
+		if strings.Contains(form.Email, "@") {
+			verificationCodeType = "email"
+			checkResult := object.CheckVerificationCode(form.Email, form.EmailCode)
+			if len(checkResult) != 0 {
+				responseText := fmt.Sprintf("Email%s", checkResult)
+				c.ResponseError(responseText)
+				return
+			}
+		} else {
+			verificationCodeType = "phone"
+			checkPhone := fmt.Sprintf("+%s%s", form.PhonePrefix, form.Email)
+			checkResult := object.CheckVerificationCode(checkPhone, form.EmailCode)
+			if len(checkResult) != 0 {
+				responseText := fmt.Sprintf("Phone%s", checkResult)
+				c.ResponseError(responseText)
+				return
+			}
+		}
+
+		// get user
+		user := object.GetUserByFields(form.Organization, form.Email)
+		if user == nil {
+			c.ResponseError("No such user.")
+			return
+		}
+
+		// disable the verification code
+		switch verificationCodeType {
+		case "email":
+			if user.Email != form.Email {
```

Comment:
Why check this? You get the user from Email right? Will the user have different Email value?